Improving Commit-Reveal and PBS for MEV Resistance.md

```diff
@@ -0,0 +1,82 @@
+---
+id: <leave blank -- will be assigned by reviewers>
+title: <Improvig Commit-Reveal Scheme and Proposer Builder Separation (PBS) for Better MEV Resistance>
+team: <Dr.Jiahua Xu, Jieliang Yin, Yebo Feng, Zhangxiang Hu>
+created: <2022-07-26>
+---
+
+# Flashbots Research Proposal - Improving Commit-Reveal Scheme and Proposer Builder Separation (PBS) for Better MEV Resistance
+ There are different kinds of Maximum Extractable Value (MEV) on DeFi like arbitrage, front-running, and sandwich attack, and one kind of MEV is that miners can arbitrarily view, censor, and reorder users' transactions before putting them on chain to earn extra profits from them at the expanse of users' interests. 
+  
+ Such kind of MEV is detrimental to users and to protect users from such MEV, different privacy-preserving techniques have been proposed to hide the details of users' transactions like the commit-reveal scheme, which allows users to submit an encrypted transaction to the chain, and reveal the content later only after it is confirmed with immutability. Compared with other full privacy-preserving techniques like zero-knowledge proof, it doesn't provide so strong privacy protection but is already enough for protecting users' transactions from MEV at lower cost and better efficiency, which is more practical and economical for most users.
+  
+ However, the current commit-reveal scheme still has unsolved problems for practical and widespread use. For each transaction, commit-reveal requires users to submit twice on chain including one encrypted and one revealed, which is not so efficient and reliable. For a better commit-reveal scheme, we design a new commit-reveal scheme with improved Proposer Builder Separation (PBS) that optimizes the whole on-chain process. Proposers reach consensus together on the encrypted transactions users commit first, then decrypt and reveal them for the builder to verify, execute and append on chain. With corresponding technical evaluation and economic analysis, we will prove that it will resist MEV better and benefit users, proposers, and builders. 
```

Comment:
Hi @jldunne Thanks for your kindly critiques. Proposers will just verify the basic format of transactions without knowing the content. Builders will view and validate the content and if they find the content is incorrect, they will prove and notify builders those invalid transactions will not be put on chain. In this way users have to ensure their content is correct because if they submit invalid transactions, they will still need to pay for transaction fees. Any critiques or better ideas, feel free to put forward. Thanks a lot!

Reply:
I see - so the idea is to make changes to PBS and/or Ethereum consensus so that it will work with this kind of commit-reveal scheme, and this is what is meant by the term "improved PBS". If that's correct, would you be able to expand on this and update this PR with some further details (high-level is fine) about potential core protocol changes that you anticipate?

Reply:
Thanks for your kindly advice @jldunne. Sure I have revised the PR and committed the changes to make it more clear with detail. Our current scheme didn't change Ethereum consensus. It could still be PoW or PoS. We just moved it before the block building. We changed the process of builder → prosper in PBS to prosper → builder to combine it with commit-reveal scheme better.

Reply:
I have also [updated](https://github.com/flashbots/mev-research/pull/78/commits/e3ea120e6db0a5ace3b1ef1ee04ef3ffcb39ba02?short_path=3882cc3#diff-3882cc357278f347aa5fdc93bc28c96ac12d48711ef8faf59dd22947f0620f4a) the proposal with more details. You can also view the [word version](https://docs.google.com/document/d/17fCUSHpRcqpWa8Czatv5x09zaXRGYmatVaxZeDOTP-0/edit?usp=sharing). Thanks a lot!

Reply:
Dear Jolene Dunne:
Hello!

Glad to meet and talk with you last week, and hugely appreciated for your golden critiques. We can adjust our research proposal to fit with PBS and MEV-Boost well as follows:

	1. Protect users’ transactions from MEV attacks by applying privacy-preserving techniques and preventing builders from seeing the details of transactions before confirming those transactions will be put on chain.

	2. Improve commit-reveal scheme between builders and proposers to protect both of them by making proposers can validate the real MEV value of the block before it’s revealed, and the body of the block can be revealed automatically once proposers accept the block. 

	3. Users and applications can set customized on-chain requirements for transactions they submit or receive including content and sequences. Builders need to follow them when appending them into the block.

	4. Besides builders, proposers are also unable to censor blocks by making all proposers can verify whether the proposed block they receive has the highest bid including transaction fees and MEV. If not they can propose the block made by other builders with the highest bid.


After making these improvements, the on-chain process could be:

	1. Users commit and submit their transactions with their requirements and bids to the mempool, and builders pick up transactions from the mempool. They are unable to see the content of transactions, but can still verify and determine what transactions will be included in their block.

	2. Proposers will choose the block with the highest bid among all proposed block from builders. After proposers make the confirmation, the winning block can be revealed automatically and sent back to the builder who propose that block.

	3. The builder will execute transactions according to uses’ requirements and construct the block accordingly. Proposers will accept the block and reach consensus to put it on chain.


Feel free to put forward any questions or critiques and we will revise accordingly. This is just the high-level thinking in general, and we will refine our solution in details and work with you to improve and achieve it if you are willing to support. Thanks a lot for your kindly consideration!


Kindly Regards,
Jieliang Yin